Sources/Basic/misc.swift

```diff
@@ -18,10 +18,10 @@ import Foundation
 ///   - path: Absolute path to the executable.
 ///   - args: The executable arguments.
 public func exec(path: String, args: [String]) throws {
-    let cArgs = CStringArray(args)
-    guard execv(path, cArgs.cArray) != -1 else {
-        throw POSIX.SystemError.exec(errno, path: path, args: args)
-    }
+    var process: Process = Process()
+    process.executableURL = path
+    process.arguments = args
+    try process.run()
```

Comment:
Two notes: 

1. This API is for replacing the current process with a new one so we shouldn't just run it as a subprocess.
2. Don't use Foundation's Process. We have our own implementation for various reasons. It is one of the thing that we'll want to implement for all platforms instead of using Foundation's.

Reply:
That isn't a great idea ... there is a bunch of complexity that is involved for process execution, particularly on Windows.  On Windows, `CreateProcess` is the closest equivalent to `exec`.

Reply:
We should conditionalize it then..